WcaOnRails/app/models/competition.rb

```diff
@@ -742,23 +742,21 @@ def self.search(query, params: {})
       competitions = competitions.where("CAST(CONCAT(year,'-',endMonth,'-',endDay) as Date) <= ?", end_date)
     end
 
-    if query.present?
-      sql_query = "%#{query}%"
-      competitions = competitions.where("id LIKE :sql_query OR name LIKE :sql_query OR cellName LIKE :sql_query OR cityName LIKE :sql_query OR countryId LIKE :sql_query", sql_query: sql_query).order(year: :desc, month: :desc, day: :desc)
+    query&.split&.each do |part|
+      like_query = %w(id name cellName cityName countryId).map { |column| column + " LIKE :part" }.join(" OR ")
+      competitions = competitions.where(like_query, part: "%#{part}%")
     end
 
-    if params[:sort].present?
-      params[:sort].split(",").each do |field|
-        order = field.start_with?("-") ? :desc : :asc
+    (params[:sort] || "-start_date").split(",").each do |field|
```

Comment:
@jfly I meant that it's not used by our code. Me and Lars just consider this to be kinda weird. Actually I thought that our API isn't that tight as it's not even documented, so making reasonable changes, especially not breaking ones is ok at the moment. If it makes sense for you I can revert this part though =) (?)